typing/typecore.ml

```diff
@@ -1249,7 +1249,10 @@ and type_pat_aux ~constrs ~labels ~no_existentials ~mode ~explode ~env
                   row_more = newvar ();
                   row_fixed = false;
                   row_name = None } in
-      unify_pat_types loc !env (newty (Tvariant row)) expected_ty;
+      (* PR#7404: allow any_extra_tag blindly, as it would not unify with
+         the abstract row variable *)
+      if constrs = None || l <> Parmatch.any_extra_tag || sarg <> None then
+        unify_pat_types loc !env (newty (Tvariant row)) expected_ty;
```

Comment:
I'd be tempted to rewrite this to:
```ocaml
if l <> Parmatch.any_extra_tag then
  unify_pat_types loc !env (newty (Tvariant row)) expected_ty
else begin
  assert (constrs <> None);
  assert (sargs = None)
end;
```
but apparently you don't want to enforce that `any_extra_tag` cannot appear in the input and be produced only by parmatch...

I feel like it would be better to enforce it, but unless someone else shares that opinion I'm fine with merging your current implementation.